ethcore/engine/src/snapshot.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2015-2019 Parity Technologies (UK) Ltd.
```

Comment:
Is there any reason why `snapshots` are part of engine?

Reply:
No good reason that I can think of. However the `Engine` trait does expose the `SnapshotComponent` trait (as a return type), which in turn pulls in a bunch of other stuff. Quite vexing but yeah, this didn't feel like the right PR to sort that out.

Reply:
Fair enough.